include/deal.II/dofs/dof_tools.h

```diff
@@ -1437,6 +1437,38 @@ namespace DoFTools
                                          const std::set<types::boundary_id> &boundary_ids = std::set<types::boundary_id>());
 
   /**
+   * Extract all degrees of freedom (DoFs) that have support only within cells,
+   * for which @p predicate is <code>true</code>. The result is returned as
+   * an IndexSet.
+   *
+   * Consider the following FE space where predicate returns <code>true</code>
+   * for all cells on the left side:
+   *
+   * @image html extract_dofs_with_support_on.png
+   *
+   * This functions will return the union of all DoFs on those cells minus
+   * DoF 11, 13, 2 and 0; the result will be <code>[9,10], 12, [14,38]</code>.
+   * On the image above the returned DoFs are separated from the rest by the
+   * red line
+   *
+   * Essentially, the question this functions answers is the following:
+   * Given a subdomain with associated DoFs, what is the largest subset of
+   * these DoFs that are allowed to be non-zero such that after calling
+   * ConstraintMatrix::distribute() the resulting solution vector will have
+   * support only within the given domain.
```

Comment:
Don't we have a constraint x_4 = x_26 meaning, x_4 will be set to x_26 or the other way around (depending on which order it is stored inside ConstraintMatrix)?